fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/NewQueryBuilder.java

```diff
@@ -598,4 +610,177 @@ private void addResourceTypeExtension(SearchQuery domainModel, List<String> reso
+            throws Exception {
+        
+        // We only need to attempt to consolidate if we have multiple query parameters for the
+        // same search parameter name. Loop through the search parameters, mapping parameter name
+        // to parameter(s) to determine if this is the case.
+        Map<String,List<QueryParameter>> consolidationMap = new HashMap<>();
+        for (QueryParameter searchParameter : searchParameters) {
+            if (Type.DATE.equals(searchParameter.getType())) {
+                consolidationMap.computeIfAbsent(searchParameter.getCode(), k -> new ArrayList<>()).add(searchParameter);
+            }
+        }
+        
+        // Now loop through the map to find any cases of same parameter specified multiple times.
+        // If found, we will attempt to consolidate. If not, we will simply process as a normal
+        // date parameter.
+        for (Map.Entry<String,List<QueryParameter>> entry : consolidationMap.entrySet()) {
+            List<QueryParameter> queryParameters = entry.getValue();
+            boolean eligibleToConsolidate = true;
+            if (queryParameters.size() == 1 || LAST_UPDATED.equals(entry.getKey())) {
+                eligibleToConsolidate = false;
+            } else {
+                // We have multiple parameters. If any of the parameters have a modifier specified, or
+                // if chain or inclusion parameter, or if multiple values, don't attempt to consolidate.
+                for (QueryParameter queryParm : queryParameters) {
+                    List<QueryParameterValue> queryParmValues = queryParm.getValues();
+                    if (queryParm.getModifier() != null || queryParm.isChained() ||
+                            queryParm.isInclusionCriteria() || queryParmValues.size() > 1) {
+                        eligibleToConsolidate = false;
+                        break;
+                    }
+                }
+            }
+            
+            if (eligibleToConsolidate) {
+                // Attempt to consolidate the upper and lower bound constraints
+                List<QueryParameter> consolidatedParms = new ArrayList<>();
+                Instant gteBound = null;
+                Instant lteBound = null;
+                Instant saBound = null;
+                Instant ebBound = null;
+                QueryParameter gteBoundParm = null;
+                QueryParameter lteBoundParm = null;
+                QueryParameter saBoundParm = null;
+                QueryParameter ebBoundParm = null;
+                for (QueryParameter queryParm : queryParameters) {
+                    QueryParameterValue queryParmValue = queryParm.getValues().get(0);
+                    Prefix prefix = queryParmValue.getPrefix();
+                    Instant valueLowerBound = queryParmValue.getValueDateLowerBound();
+                    Instant valueUpperBound = queryParmValue.getValueDateLowerBound();
```

Comment:
Should this be "getValueDateUpperBound()"?

Reply:
yes, updated - nice catch